### ISSUE TITLE
js-crawler seems to crawl the same url multiple times

### ISSUE THREAD

**Issue by JulienBonnin-ABTasty** [antivanov/js-crawler#21]
Hello,

I am using js-crawler and it's awesome, but I have an issue when crawling a small website.

Here is my code

```
var crawler = new Crawler().configure({
    ignoreRelative: false,
    depth: Number.POSITIVE_INFINITY,
    shouldCrawl: function(url) {
        return url.indexOf('.pdf') < 0 && url.indexOf(host) >= 0;
    }
});

var crawledPages = {};

crawler.crawl({
    url: entryPage,
    success: function(page) {
        console.log('success crawling -> ', page.url);
        crawledPages[page.url] = page.body;
    },
    failure: function(page) {
        console.log("ERROR crawling url => ", page);
    },
    finished: function(crawledUrls) {
        console.log('length of pages crawled is => ', crawledUrls.length); // output 177
        console.log('length of different urls => ', Object.size(crawledPages)); // output 22
        htmlStripper.strip(crawledPages);
    }
});
```

As you can see, the output of my console.log in the "finished" callback are different. Why so ? 
It is not a major issue because I have my 22 pages in my crawledPages variable, but how can I avoid crawling the same url twice ?

Thanks :)

**Comment by antivanov** [comment]
Thank you for filing an issue, I will take a look.

**Comment by antivanov** [comment]
There was bug in the code that tracks the urls currently being crawled, that should have been resolved now, the new version has been published to NPM 0.3.10

I plan to add more automated end-to-end tests, improve the quality of the crawler and publish a new major release in a few month time.

Please, check if the problem has been resolved.

**Comment by ivan-kleshnin** [comment]
I have the same problem with `"js-crawler": "^0.3.21",`

```js
200 - http://localhost:8080/tutorials/async-programming/callbacks/challenges/8.2.tsk/
200 - http://localhost:8080/tutorials/async-programming/callbacks/challenges/8.2.tsk/
200 - http://localhost:8080/tutorials/async-programming/callbacks/challenges/8.2.tsk/
```

Maybe the presence of `shouldCrawl` disables *visited* check or something. Sorry, have no time to investigate...
